### PR TITLE
Instantiate rework with new keywords and fewer functions.

### PR DESCRIPTION
**Pull request by DylanScottCarroll**
I reworked all of the instantiation behavior.

I removed the keyword _target_ and replaced it with the keywords _instance_, _partial_, and _fetch_.
This also replaces the functionality of sk.fetch() and sk.instantiate_partial(), as now the behavior of instantiation is determined in the config rather than in the code.

I also removed sk.instantiate_all(), by allowing sk.instantiate() to take multiple configs as positional arguments and return a list of instantiated objects.

I also updated the examples to use these new keywords. I tested with them and they all seem to work to my eyes.

### CODE REVIEW

**Comment by DylanScottCarroll** [comment]
There are two design decisions I made here that I wasn't fully sure about.

The first one was what exactly the keywords are. I ended up going with _instance_, _partial_, and _fetch_, but I think these could be anything else as well so long as it makes sense.

Secondly, I added a check for the case that there is a _fetch_ keyword, but the user supplies keyword arguments (either via the config or the instantiate function call. When fetching, no arguments are actually ever supplied to anything. It seems to me that doing this would usually be an error, which is why I set it to raise an error. Though, I can imagine there might be use-cases where a _fetch_ config might have other values even if they are extraneous.

File: skeletonkey/instantiate.py [hunk 1]
```diff
@@ -24,143 +28,151 @@ def import_target(class_string: str) -> Type[Any]:
...
+    
+    if len(configs) == 1:
+        return _instantiate_single(configs[0], instance_keyword, partial_keyword, fetch_keyword, _instantiate_recursive, **kwargs)
+    
+    else:
+        return [_instantiate_single(config, instance_keyword, partial_keyword, fetch_keyword, _instantiate_recursive, **kwargs)
+                for config in configs]
+            
+
+def _instantiate_single(config: Config,
```
Comment (sizemore0125) [inline skeletonkey/instantiate.py:68]: Not consistent styling with the rest of skeletonkey. Please do something like:

def _instantiate_single(
    config: Config,
    instance_keyword: str,
    partial_keyword: str,
          ...
    **extra_kwargs,
) -> Any:

File: example/partial_instantiate_example/main.py [hunk 2]
```diff
@@ -30,13 +30,13 @@ def __call__(self, x):
...
 def main(config):
-    partial_nn = skeletonkey.instantiate_partial(config.neural_network)
+    partial_nn = skeletonkey.instantiate(config.neural_network)
     print(f"Type of partial_nn: {type(partial_nn)}")
     # >>> Type of partial_nn: <class 'functools.partial'>
 
     epic_model = partial_nn(input_size=28**2, output_size=10)
     print(f"Type of epic_model: {type(epic_model)}")
-    # >>> Type of partial_nn: <class 'functools.partial'>
+    # >>> Type of epic_model: <class 'main.SuperCoolNeuralNetwork'>
```
Comment (sizemore0125) [inline example/partial_instantiate_example/main.py:39]: Is this true? I think it would still be functools.partial.

File: skeletonkey/config.py [hunk 3]
```diff
@@ -100,6 +98,7 @@ def __getattr__(self, name):
             raise AttributeError(message + f" Please specify '{name}' in your config yaml.")
     
     def instantiate(self, **kwargs):
+        from .instantiate import instantiate
```
Comment (sizemore0125) [inline skeletonkey/config.py:101]: Why did you do this? Shouldn't this be defined at the top of this file?

File: skeletonkey/instantiate.py [hunk 4]
```diff
@@ -24,143 +28,151 @@ def import_target(class_string: str) -> Type[Any]:
...
 
 
-def instantiate_partial(config, target_keyword=TARGET_KEYWORD,_instantiate_recursive=True, **kwargs) -> Any:
-    """
-    Instantiate a functools.partial object using a Config object. The Config
-    object should contain the key "_target_" to specify the class to instantiate.
-
-    You need to call the partial object with the remaining parameters to
-    fully instantiate.
+def instantiate(*configs:Iterable[Config],
```
Comment (sizemore0125) [inline skeletonkey/instantiate.py:31]: Not proper styling for skeletonkey. Check later comment.

**Review by sizemore0125 — CHANGES_REQUESTED** [review]
Overall good. Please check the comments for changes.

File: skeletonkey/instantiate.py (hunk 1 above)
Comment (DylanScottCarroll) [reply skeletonkey/instantiate.py]: Fixed

File: example/partial_instantiate_example/main.py (hunk 2 above)
Comment (DylanScottCarroll) [reply example/partial_instantiate_example/main.py:39]: This is true. When you call a functools.partial, the function (or class constructor in this case) it contains is called and the return value of that is the output.

File: skeletonkey/config.py (hunk 3 above)
Comment (DylanScottCarroll) [reply skeletonkey/config.py:101]: This prevents circular imports by waiting for instantiate to be fully resolved before importing it.

File: skeletonkey/instantiate.py (hunk 4 above)
Comment (DylanScottCarroll) [reply skeletonkey/instantiate.py]: Fixed

**Pull request merged**